src/prefect/core/task.py

```diff
@@ -21,6 +21,33 @@
 VAR_KEYWORD = inspect.Parameter.VAR_KEYWORD
 
 
+def _validate_init_signature(init: Callable) -> None:
+    init_sig = inspect.getfullargspec(init)
+    reserved_kwargs = [
```

Comment:
This feels fragile -- is it possible to read this list directly from `Task.__init__` rather than hardcode?

Reply:
Unfortunately not - it creates a chicken and the egg problem. This validate lives in the class that _creates_ the Task object. I tried it but resorted to the hard coded list

Reply:
Understood. Perhaps we could unit test the list to make sure it at least is compliant?

Reply:
Ok yea good idea - I’ll add that!

Reply:
Awesome - added a test for this!